common/rect.h

```diff
@@ -109,91 +143,104 @@ static inline Point operator*(double multiplier, const Point &p) { return Point(
  * When writing code using our Rect class, always keep this principle in mind!
 */
 struct Rect {
-	int16 top, left;		///< The point at the top left of the rectangle (part of the rect).
-	int16 bottom, right;	///< The point at the bottom right of the rectangle (not part of the rect).
+	int16 top, left;		/*!< The point at the top left of the rectangle (part of the Rect). */
+	int16 bottom, right;	/*!< The point at the bottom right of the rectangle (not part of the Rect). */
 
 	Rect() : top(0), left(0), bottom(0), right(0) {}
+	/**
+     * Create a rectangle with width @p w and height @p h.
+     */
 	Rect(int16 w, int16 h) : top(0), left(0), bottom(h), right(w) {}
+	/**
+     * Create a rectangle between the points defined with @p x1, @p y1, @p x2, and @p y2.
+	 * Additionally, check whether it is a valid rectangle.
+     */
 	Rect(int16 x1, int16 y1, int16 x2, int16 y2) : top(y1), left(x1), bottom(y2), right(x2) {
 		assert(isValidRect());
 	}
+	/**
+     * Create a rectangle with width @p w and height @p h.
+     */
 	bool operator==(const Rect &rhs) const { return equals(rhs); }
+    /**
+     * Create a rectangle with width @p w and height @p h.
+     */
 	bool operator!=(const Rect &rhs) const { return !equals(rhs); }
 
-	int16 width() const { return right - left; }
-	int16 height() const { return bottom - top; }
+	int16 width() const { return right - left; }  /*!< Compute the width of a rectangle. */
```

Comment:
I would not say `compute` here, but just `return`.

It's an implementation detail that the rectangle stores internally the left and right position rather than the left position and the width. The two would be equivalent for the developer using a rectangle.